crates/router/src/core/payments.rs

```diff
@@ -746,6 +761,16 @@ where
             }
             _ => (router_data, should_continue_payment),
         },
+        Some(api_models::payments::PaymentMethodData::Wallet(_)) => {
+            if connector.connector_name.to_string() == *"trustpay" {
+                (
+                    router_data.preprocessing_steps(state, connector).await?,
+                    false,
+                )
```

Comment:
why are we using preprocessing steps? why not use the authorize api itself for applepay payments?

Reply:
We get session tokens and not exactly authroizing payments , so we are going with pre processing steps.